language/move-prover/diagen/src/lib.rs

```diff
@@ -0,0 +1,230 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use log::{error, info};
+use regex::Regex;
+use serde::{self, Deserialize, Serialize};
+use std::{
+    collections::{btree_map::Entry, BTreeMap, BTreeSet, VecDeque},
+    env, fs,
+    io::{self, Write},
+    path::{Path, PathBuf},
+    process::{Command, Stdio},
+};
+
+/// Options passed into the documentation generator.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(default, deny_unknown_fields)]
+pub struct DiagenOptions {
+    pub generate_dot: bool,
+    pub generate_pdf: bool,
+    pub generate_svg: bool,
+    pub generate_global_graph: bool,
+}
+
+impl Default for DiagenOptions {
+    fn default() -> Self {
+        Self {
+            generate_dot: false,
+            generate_pdf: false,
+            generate_svg: false,
+            generate_global_graph: false,
+        }
+    }
+}
+
+pub fn move_files_in(path: &PathBuf) -> Vec<PathBuf> {
```

Comment:
I think the tool should not generate diagrams for every .move file in the folder when called from docgen. Rather you should have an entry point into this crate where it generates a diagram file at a specific file location,

Reply:
Refactored accordingly.